### DISCUSSION TITLE
Reusing ; for surround

### DISCUSSION THREAD

**Discussion by dusty-phillips** [LazyVim/LazyVim#2547] [Q&A] [answered]
I use mini-surround keybindings a lot, so I'm not fond of the two-character `gs` keybinding. I decided to remap it to `;` since "find next" is handled by the `f` key with flash.

So I remapped mini.surround as follows:

```
return {
  "echasnovski/mini.surround",
  opts = {
    mappings = {
      add = ";;",
      delete = ";d",
      find = ";f",
      find_left = ";F",
      highlight = ";h",
      replace = ";r",
      update_n_lines = ";n",
    },
  },
}
```

This works fine if I press the two keys quickly enough. However, a single `;` press causes the original `find next` semicolon functionality to happen instead.

I thought I could resolve this by adding `vim.keymap.del("n", ";")` to my `keymaps.lua`. However, that seems to disable the `mini.surround` keymaps as well.

Interestingly, checkhealth still indicates that there are conflicting `;` keymaps even though at this point, `;` doesn't do *anything* in my config:

```
 WARNING conflicting keymap exists for mode **"n"**, lhs: **";d"**
- rhs: ` `
- WARNING conflicting keymap exists for mode **"n"**, lhs: **";f"**
- rhs: ` `
- WARNING conflicting keymap exists for mode **"n"**, lhs: **";h"**
- rhs: ` `
- WARNING conflicting keymap exists for mode **"n"**, lhs: **";r"**
- rhs: ` `
- WARNING conflicting keymap exists for mode **"n"**, lhs: **";F"**
```

Is there a more correct way to remap the mini.surround keys and/or to disable the existing ; key? Ideally, I'd get a which-key menu when I hit the `;` so I can see what the surround keystrokes are.

**Comment by dpetka2001** [comment]
I tried your exact config in a fresh LazyVim starter template and this is what I get when I press `;`

![2024-02-13_21-19](https://github.com/LazyVim/LazyVim/assets/12776461/38ad7340-61eb-4dd2-b177-b3908e61609a)

**Reply by dusty-phillips** [reply]
Reliably? I seem to get that view 20% of the time and the standard ; behaviour 80% of the time. 🤨 

Just noticed that `:checkhealth which-key` shows conflicts in the 80% time and no conflicts when it's working as expected. Race condition of some sort?

**Reply by dpetka2001** [reply, marked as answer]
Yes I did notice it some times. But most of the times I was observing the desired behavior. Never the less, I ended up with following
```lua
  {
    "folke/flash.nvim",
    opts = {
      modes = {
        char = {
          keys = { "f", "F", "t", "T", [";"] = "'", "," },
        },
      },
    },
  },
```
and now I'm observing only your desired behavior after I restarted Neovim 10-15 times. Test it out to see if it fits your needs. In `flash` I replaced `;` with `'` (single quote), but feel free to replace it with whatever suits you.

**Reply by dusty-phillips** [reply]
I just left `;` and `,` out of the list and it appears working fine across several restarts. thanks!
